src/wallet/wallet.ts

```diff
@@ -143,16 +186,45 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start() {
+  async start({ pinCode }) {
     if (!this.seed) {
       throw new Error('Seed should be in memory when starting the wallet.');
     }
 
     this.setState(walletState.LOADING);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
     const xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    const authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(this.seed, {
+      passphrase: this.passphrase,
+      networkName: this.network.name,
+    });
+
+    const walletId = walletUtils.sha256d(xpub, 'hex');
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+
+    // prove we own the xpubkey
+    const xpubAccountPath = walletUtils.deriveXpriv(xpriv, '0\'')
+    const address = xpubAccountPath.publicKey.toAddress(this.network.getNetwork()).toString();
+    const message = new bitcore.Message(String(timestampNow).concat(walletId as string).concat(address));
+    const xpubkeySignature = message.sign(xpubAccountPath.privateKey);
+
+    // prove we own the auth_xpubkey
+    const authDerivedPrivKey = HathorWalletServiceWallet.deriveAuthXpriv(xpriv);
+    const authAddress = authDerivedPrivKey.publicKey.toAddress(this.network.getNetwork());
+
+    const authMessage = new bitcore.Message(String(timestampNow).concat(walletId as string).concat(authAddress));
+    const authXpubkeySignature = authMessage.sign(authDerivedPrivKey.privateKey);
+
     const xpubChangeDerivation = walletUtils.xpubDeriveChild(xpub, 0);
     const firstAddress = walletUtils.getAddressAtIndex(xpubChangeDerivation, 0, this.network.name);
+
     this.xpub = xpub;
+    this.authXpriv = authDerivedPrivKey;
```

Comment:
You are setting to the `authXpriv` a private key and not an xpriv, is that correct?

Reply:
You are right, refactored methods at 76913ffaad9f3379fa19ba8bf1fa53ae3a3f05c1